### PR TITLE
fix(authentication ): add validation in authentication interface

### PR DESCRIPTION
**Pull request by hamzamahmood**
The following PR adds the Validate method in the authentication interface.

Closes# 13

### CODE REVIEW

File: src/main/java/io/apimatic/coreinterfaces/authentication/Authentication.java [hunk 1]
```diff
@@ -15,4 +15,9 @@ public interface Authentication {
      * @return the authenticated request
      */
     Request apply(Request httpRequest);
+    
+    /**
+     * Validate the authentication on the httpRequest
```
Comment (sufyankhanrao) [inline src/main/java/io/apimatic/coreinterfaces/authentication/Authentication.java:20]: Description should look something like "validates the auth params for the Http Request."

**Review by sufyankhanrao — CHANGES_REQUESTED** [review]
Please incorporate the above suggestion.

File: src/main/java/io/apimatic/coreinterfaces/authentication/Authentication.java (hunk 1 above)
Comment (hamzamahmood) [reply src/main/java/io/apimatic/coreinterfaces/authentication/Authentication.java]: The comment has been updated

**Review by sufyankhanrao — APPROVED** [review]
Approved

**Pull request merged**